python_modules/dagster/dagster/_core/definitions/external_asset_graph.py

```diff
@@ -122,106 +93,216 @@ def from_repository_handles_and_external_asset_nodes(
         repo_handle_external_asset_nodes: Sequence[Tuple[RepositoryHandle, "ExternalAssetNode"]],
         external_asset_checks: Sequence["ExternalAssetCheck"],
     ) -> "ExternalAssetGraph":
-        upstream: Dict[AssetKey, AbstractSet[AssetKey]] = {}
-        source_asset_keys: Set[AssetKey] = set()
-        partitions_defs_by_key: Dict[AssetKey, Optional[PartitionsDefinition]] = {}
-        partition_mappings_by_key: Dict[AssetKey, Dict[AssetKey, PartitionMapping]] = defaultdict(
-            defaultdict
-        )
-        group_names_by_key = {}
-        freshness_policies_by_key = {}
-        auto_materialize_policies_by_key = {}
-        backfill_policies_by_key = {}
-        keys_by_atomic_execution_unit_id: Dict[str, Set[AssetKeyOrCheckKey]] = defaultdict(set)
         repo_handles_by_key = {
             node.asset_key: repo_handle
             for repo_handle, node in repo_handle_external_asset_nodes
             if not node.is_source or node.is_observable
         }
-        job_names_by_key = {
-            node.asset_key: node.job_names
-            for _, node in repo_handle_external_asset_nodes
-            if not node.is_source or node.is_observable
+
+        # Split the nodes into materializable, observable, and unexecutable nodes. Observable and
+        # unexecutable nodes represent both source and external assets.
```

Comment:
Just to leave no doubt, this means that a `SourceAsset` in a code location gets serialized as an `ExternalAssetNode` and this in basically an unfortunate naming collision because of the entire External family of classes

Reply:
exactly

Reply:
Cool. Let's leave a comment to that effect

Reply:
updated comment